File: Technical/scripts/shell-v1.R
library(rjson)
library(optparse)

main <- function(ac, av) {
  optionsList <- list(
    make_option(c("-s", "--script"), dest="script", type="character", help="[Required] User script code to run"),
    make_option(c("-p", "--params"), dest="params", type="character", default="{}", help="Configuration parameters in JSON format"),
    make_option(c("-i", "--input"), dest="inputs", type="character", default="", help="Input files separated by a comma")
  );
  parser <- OptionParser(usage="Usage: %prog --script <path_to_user_script> [options]", option_list = optionsList, description="Borderline R wrapper script");
  args <- parse_args(parser, args = av);
  
  if (is.null(args$script)) {
    print_help(parser);
    quit(status = 1);
  }
  
  #Parse JSON input to native R data structure
  args$params <- fromJSON(args$params);
  
  #Parse Input file list to Native R vector
  args$inputs <- unlist(strsplit(c(args$inputs), ',', fixed=TRUE));
  
  source(args$script);
  borderlineJob(args$params, args$inputs);
}


#Getting command line arguments
cmdArgs <- commandArgs(trailingOnly = TRUE);
##Passing that to a c-like main function
main(length(cmdArgs), cmdArgs)

File: packages/borderline-documentation/Technical/scripts/shell-v1.py
#!/bin/python2.7

import sys, os, imp
import argparse
import json

def main():
    parser = argparse.ArgumentParser(description='Borderline R wrapper script')
    parser.add_argument("-i", "--input", dest="inputs", action="append", help="Input file")
    parser.add_argument("-s", "--script", dest="script", required=True, help="User script file to run")
    parser.add_argument("-p", "--params", dest="params", default='{ "isDefault": true }', help="Configuration parameters in JSON format")
    args = parser.parse_args()

    jsonParams = json.loads(args.params)
    script = imp.load_source('', args.script)
    script.borderline_job(jsonParams, args.inputs)


if __name__ == "__main__":
    main()
